openedx/core/djangoapps/agreements/views.py

```diff
@@ -85,12 +94,24 @@ def get(self, request, course_id):
                 status=status.HTTP_404_NOT_FOUND,
             )
 
-        created_at = str(signature.created)
+        serializer = IntegritySignatureSerializer(signature)
+        return Response(serializer.data)
 
-        data = {
-            'username': username,
-            'course_id': course_id,
-            'created_at': created_at,
-        }
+    def post(self, request, course_id):
```

Comment:
I wasn't able to gate access to the API by using the waffle flag in urls.py (tests were failing because there were database access attempts to the waffle flag table during initialization which wasn't allowed). I gated access to the get request by returning a 404 if the waffle flag wasn't enabled, should something similar be done here? I'm also open to more ideas if we think that access should be gated in urls.py

Reply:
Oh, yeah, that makes sense. I can add that.